### ISSUE TITLE
Support `axis=None` in `argsort` etc

### ISSUE THREAD

**Issue by agoose77** [scikit-hep/awkward#1855]
### Description of new feature

Currently we don't support this argument value.

**Comment by jpivarski** [comment]
There's a lot of history with sorting and `axis`, particularly the question, "what _should_ it do?" Python and I think it's most natural for list and records above the level specified by the `axis` to be unbroken, thanks to an ordering defined on whole list/record objects, but NumPy always sorts numbers within horizontal or vertical slices of 2 or more dimensional arrays.

```python
>>> sorted([[10, 2, 3], [4, 50, 60]])
[[4, 50, 60], [10, 2, 3]]
>>> np.sort([[10, 2, 3], [4, 50, 60]], axis=0)
array([[ 4,  2,  3],
       [10, 50, 60]])
>>> np.sort([[10, 2, 3], [4, 50, 60]], axis=1)
array([[ 2,  3, 10],
       [ 4, 50, 60]])
```

Awkward Array does what NumPy does, which gets even more surprising when variable-length lists and missing values are included. Fortunately, the sorter semantics follows reducer semantics in how horizontal and vertical slices are identified by `axis`, so we don't have to make something up.

`axis=None` should be simpler: everything gets broken down and sorted together. In Awkward, this would just be a `completely_flatten` followed by one sort.

```python
>>> np.sort([[10, 2, 3], [4, 50, 60]], axis=None)
array([ 2,  3,  4, 10, 50, 60])
```

Issues/PRs that have touched on this before:

* #74
* #168
* #506 (correct behavior surprised a user)
* #523 (implementation issue)
* #524 (implementation issue)
* #926 (documentation)
* #946
* #1072

**Comment by agoose77** [comment]
Agreed, it's just a missing case for us to add. I'm holding off here, because it's related to #1362 and others
